utilities/copyoffload_migration.py

```
@@ -887,6 +888,186 @@ def _get_populate_pod_logs(
+    Args:
+        plan (Plan): The Plan CR resource with completed migration status.
+
+    Returns:
+        list[Any]: All tasks from DiskTransfer pipeline steps across all VMs.
+
+    Raises:
+        ValueError: If plan has no migration status, no VMs, or no DiskTransfer tasks.
+    """
+    plan_status = plan.instance.status
+    if plan_status is None:
+        raise ValueError(f"Plan '{plan.name}' has no status")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)